Switch travis scripts to use beaver

mihails-strasuns-sociomantic opened the pull request.

mihails-strasuns-sociomantic committed:
Add beaver submodule version 0.1.2

mihails-strasuns-sociomantic committed:
Switch travis scripts to use beaver

mihails-strasuns-sociomantic commented on .travis.yml:
```diff
@@ -1,23 +1,35 @@
+install: beaver install
 
-script: ci/travis-script.sh
+script: beaver run ci/travis.sh
```
@leandro-lucarella-sociomantic how does one pass env vars like `DC` through beaver here?

mihails-strasuns-sociomantic replied:
(right now it is confusingly green because it defaults to dmd1 + production in all matrix branches)

mihails-strasuns-sociomantic replied:
Ping @leandro-lucarella-sociomantic

leandro-lucarella-sociomantic replied:
`BEAVER_DOCKER_VARS="CC DIST"`

I know docs are still pretty bad, but there is something there: https://github.com/sociomantic-tsunami/beaver#docker

leandro-lucarella-sociomantic reviewed (approved):
You might want to use beaver v0.1.2 now. The rest LGTM with the necessary adjustments.

mihails-strasuns-sociomantic committed:
Remove env var defaults

Those were added as extra convenience to run CI scripts on dev machines
but that introduced additional point of failure in actual CI - if env
var passing get broken, CI build will still succeed with defaults.

mihails-strasuns-sociomantic commented:
Fixed env var passing and removed default values to ensure build does not pass by an accident.
